### PR TITLE
Merge internal changes and add some cleanup after #584.

### PR DESCRIPTION
**Pull request by mingxingtan**

### CODE REVIEW

File: efficientdet/coco_metric.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ def reset_states(self):
     self.annotation_id = 1
     self.category_ids = []
     self.metric_values = None
+    self.coco_gt = None
```
Comment (fsx950223) [inline efficientdet/coco_metric.py:80]: Never used.

File: efficientdet/det_model_fn.py [hunk 2]
```diff
@@ -456,35 +454,53 @@ def _model_outputs(inputs):
...
-          nms_scores,
-          nms_classes,
-      ]
-      detections = tf.stack(detections, axis=-1, name='detnections')
-      kwargs['detections_bs'] = detections
+      if params['nms_configs'].get('pyfunc', True):
+        detections_bs = []
+        for index in range(kwargs['boxes'].shape[0]):
+          nms_configs = params['nms_configs']
+          detections = tf.py_func(
```
Comment (fsx950223) [inline efficientdet/det_model_fn.py:461]: numpy_function or py_function

File: efficientdet/iou_utils.py [hunk 3]
```diff
@@ -38,17 +38,20 @@ def _get_grad_v(height, width):
...
 
     def _grad_v_graph(dv, variables):
+      """Grad for graph mode."""
       gdw = dv * 8 * arctan * height / (math.pi**2)
       gdh = -dv * 8 * arctan * width / (math.pi**2)
       return [gdh, gdw], tf.gradients(v, variables, grad_ys=dv)
 
-    return v, _grad_v if tf.compat.v1.executing_eagerly_outside_functions(
-    ) else _grad_v_graph
+    if tf.compat.v1.executing_eagerly():
```
Comment (fsx950223) [inline efficientdet/iou_utils.py:52]: It should be ```tf.compat.v1.executing_eagerly_outside_functions```.

File: efficientdet/keras/train_lib.py [hunk 4]
```diff
@@ -337,19 +334,21 @@ def __init__(self, iou_loss_type, min_level, max_level, num_scales,
...
-    input_anchors = BoxList(
-        tf.tile(self.input_anchors.boxes,
-                [box_outputs.shape[0] // self.input_anchors.boxes.shape[0], 1]))
+    anchor_boxes = tf.tile(
+        self.input_anchors.boxes,
+        [box_outputs.shape[0] // self.input_anchors.boxes.shape[0], 1])
     num_positives, box_targets = y_true
-    box_outputs = self.box_coder.decode(box_outputs, input_anchors)
-    box_targets = self.box_coder.decode(box_targets, input_anchors)
+    print('box_outputs=', box_outputs.shape)
```
Comment (fsx950223) [inline efficientdet/keras/train_lib.py:344]: logging.info?

File: efficientdet/coco_metric.py [hunk 5]
```diff
@@ -148,19 +149,19 @@ def update_state(self, groundtruth_data, detections):
         [image_id, x, y, width, height, score, class].
     """
     for i in range(len(detections)):
+      detections_i = detections[i]
```
Comment (LucasSloan) [inline efficientdet/coco_metric.py:152]: It looks like i is only used to index into detections and groundtruth.  Maybe replace the for loop with:

```
for d, gt in zip(detections, groundtruth:
```

**Pull request merged**